docs/site/tutorials/authentication/Quick-Authentication-Tutorial-Session-Tokens.md

```diff
@@ -0,0 +1,315 @@
+
+// In the handle function
+
+    const route = this.findRoute(request);
+    await this.authenticateRequest(request);    // ADD THIS LINE
+    const args = await this.parseParams(request, route);
+```
+
+## Create a Basic authentication strategy
+
+The AuthenticationStrategy is where the work really happens. It will extract the
+authentication details from the HTTP request, and check them against the
+database.
+
+Make a new folder `src/strategies`
+
+Create the file `src/strategies/basic-strategy.ts` with the following content:
+
+```ts
+import {AuthenticationStrategy} from '@loopback/authentication';
+import {UserProfile, securityId} from '@loopback/security';
+import {inject} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {HttpErrors, Request} from '@loopback/rest';
+import {UserRepository} from '../repositories';
+
+export interface Credentials {
+  clientKey: string;
+  clientSecret: string;
+}
+
+export class BasicAuthenticationStrategy implements AuthenticationStrategy {
+  name = 'BasicStrategy';
+
+  constructor() {}
+  //@repository(UserRepository) public userRepository: UserRepository,
+
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    const credentials: Credentials = this.extractCredentials(request);
+
+    // Initial mock implementation
+    // TODO Replace this with your own implementation
+    if (
+      credentials.clientKey === 'testuser' &&
+      credentials.clientSecret === 'password123'
+    ) {
+      return {name: 'testuser', [securityId]: 'abc123'};
+    } else {
+      throw new HttpErrors.Unauthorized('The credentials are not correct.');
+    }
+
+    // You should implement your own checks here
+    // 1. Find the user in your database
+    // 2. Ensure credentials.clientSecret matches the user's hashed password
+    // 3. Return the user if authenticated
+    //    Otherwise return an error (cancel the request),
+    //    or return undefined (let your controllers handle the missing user)
+
+    // Here is a more complete example you could start from
+    /*
+    const foundUser = await this.userRepository.findOne({
+      where: {username: credentials.clientKey},
+      // or alternatively
+      where: {email: credentials.clientKey},
+    });
+
+    if (!foundUser) {
+      throw new HttpErrors.NotFound(
+        `User with clientKey ${credentials.clientKey} not found.`,
+      );
+    }
+
+    // Using the password hasher from https://bit.ly/34q3IJp
+    const passwordMatched = await this.passwordHasher.comparePassword(
+      credentials.clientSecret,
+      foundUser.password,
+    );
+
+    if (passwordMatched) {
+      // We must at least return a UserProfile with a securityId
+      return {email: foundUser.email, [securityId]: foundUser.id};
+      // Or we can return the full User object, with the securityId added to it
+      // This will give the controllers more user properties to play with
+      foundUser[securityId] = foundUser.id;
```

Comment:
I see now that this upsets TypeScript. In our app we just `@ts-ignore` it!

Is there any nice way to do this?

(We like to pass the user directly to our controllers.)

Reply:
I was half-way trying out your commented portion :) 
I used the snippet below instead: (my `User` might be different than yours)
```
return {[securityId]: foundUser.email, id: foundUser.email}
```

Reply:
Yeah that looks good. I should not be trying to return a `User` object in this tutorial.